pinot-query-planner/src/main/java/org/apache/pinot/calcite/sql/fun/PinotOperatorTable.java

```diff
@@ -67,10 +67,11 @@
  */
 @SuppressWarnings("unused") // unused fields are accessed by reflection
 public class PinotOperatorTable implements SqlOperatorTable {
-  private static final Supplier<PinotOperatorTable> INSTANCE = Suppliers.memoize(PinotOperatorTable::new);
+  /// Lookup map containing instances of this class keyed by configuration.
+  private static final Map<Map<String, String>, PinotOperatorTable> INSTANCES = new ConcurrentHashMap<>();
```

Comment:
Using maps as keys may be expensive. Given that this has to be called at least once for each query, we may need to reconsider this implementation.

I was expecting something like named libraries. A library would be a named collection of operators, and each query could decide which library should be used (with a default defined at the cluster level). Then we would have a Map<String, PinotOperatorTable> here.

In the future we can modify the discovery method to be able to annotate functions to be visible for different libraries, but right now we can modify this code to use this named library system and just statically support two different libraries: one where `now` returns long and another where `now` returns timestamps.